src/provider/model/APIHandler.ts

```diff
@@ -102,11 +106,20 @@ export class APIHandler<T extends Enum, E extends EventSpecification> {
     }
 
     // TODO?: Remove the need for this any by defining connection payload type?
-    private onConnection(app: Identity, payload?: any): void {
+    private onConnectionHandler(app: Identity, payload?: any): void {
         if (payload && payload.version && payload.version.length > 0) {
             console.log(`connection from client: ${app.name}, version: ${payload.version}`);
         } else {
             console.log(`connection from client: ${app.name}, unable to determine version`);
         }
+        // The 'onConnection' callback fires *just before* the channel is ready.
```

Comment:
This will be a "4th service" task, I think. Given the APIHandler stuff is still in flux even now, doesn't really seem stable enough to break out just yet.

I'd say shelve this for now, and we'll write up when we have another service to plug it into.